src/Microsoft.ML.TimeSeries/STL/Loess.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Loess is short for Robust Locally Weighted Regression and Smoothing Scatterplots.
+    /// </summary>
+    internal class Loess
+    {
+
+        private const double NumericalThreshold = 1.0e-10;
+
+        /// <summary>
+        /// the ratio to determine the local region
+        /// </summary>
+        private readonly int _r;
+
+        private readonly bool _isTemporal;
+
+        /// <summary>
+        /// key is the index of the given point, value is the corresponding neighbors of the given point.
+        /// </summary>
+        private readonly Dictionary<int, LocalRegression> _neighbors;
+
+        private IReadOnlyList<double> _x;
+        private IReadOnlyList<double> _y;
+        private readonly int _length;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Loess"/> class.
+        /// constructing the least square algorithm
+        /// </summary>
+        /// <param name="xValues">the corresponding x-axis value</param>
+        /// <param name="yValues">the corresponding y-axis value</param>
+        /// <param name="isTemporal">if the regression is considered to take temporal information into account. in general, this is true if we are regressing a time series, and false if we are regressing scatter plot data</param>
+        public Loess(IReadOnlyList<double> xValues, IReadOnlyList<double> yValues, bool isTemporal)
+        {
```

Comment:
This and the other constructor are very similar. Is it possible to merge them, and make `int r` either an optional parameter with a default parameter (`int r = -1` or something like that) or a nullable optional parameter (`int? r = null`) in order to have only one constructor?

Reply:
Refactored to merge these two constructors.

---
In reply to: [441196395](https://github.com/dotnet/machinelearning/pull/5202#discussion_r441196395) [](ancestors = 441196395)